### DISCUSSION TITLE
PowerShell winget

### DISCUSSION THREAD

**Discussion by Tiger348** [microsoft/winget-cli#2388] [General]
A kind member suggested I step through the trouble shooting page. I was unable to resolve my issue.

In sort when using cmd prompt , When I type winget without the fle extention ".exe" . It functions as it should. But when using PowerShell I have to type winget.exe in order to function as normal.

If I type winget in isolation it prompt windows via popup "How do yu want to open this file" image attached.

This may should trivial but I'd like to resolve if possible.

Any guidence is much appreciated.

Tiger348

![Winget  exe Association](https://user-images.githubusercontent.com/109897924/181486372-3478cb95-8972-4a99-91c0-db79e648424c.png)
![winget issue 002](https://user-images.githubusercontent.com/109897924/181486381-6030a1ba-b53d-4843-8537-1d8aab1fbf7f.png)
![winget Issue](https://user-images.githubusercontent.com/109897924/181486383-86d3df62-840f-4cae-8f75-eb015947c499.png)

**Comment by florelis** [comment]
Maybe `winget` and `winget.exe` are resolving to different things on PS? What do you get if you run `Get-Command winget` and `Get-Command winget.exe`?

**Reply by Tiger348** [reply]
Thank you for your response.

The image attached provides results....points to two (2) different locations?

Tiger 348
![Get-Command PS Results](https://user-images.githubusercontent.com/109897924/181702836-53ebc044-3623-4cc3-a116-2d542e7031af.png)

**Reply by florelis** [reply]
I don't think the file `C:\Windows\System32\winget` should be there. The one you should be calling is the one in `%LOCALAPPDATA%`. I'm not sure why PS and cmd are resolving `winget` differently, but when you try to start it from PS, since it does not have a file extension (like `.exe` or `.txt`) Windows doesn't know what to do with the file and asks you what program to use.

You should remove `C:\Windows\System32\winget`. I think that should be safe, but you can inspect the file first to be sure that you don't need it (e.g., look at the file properties or open it with Notepad); you can also rename/move it instead of removing it in case you need to restore it later.

**Reply by Tiger348** [reply]
Hi there,
I'd like to thank you for your advise....."it worked"

Can I impose with another question. Do you know how I change setting in PowerShell to change the colour of the progress bar as per screen shot attached?
Tiger348
![winget Progress Bar Colour](https://user-images.githubusercontent.com/109897924/181870698-b4b89dce-7cb0-401e-898b-26c7611f815d.png)

**Reply by florelis** [reply]
For the progress bar in winget you can use the [`progressBar` setting](https://github.com/microsoft/winget-cli/blob/master/doc/Settings.md#visual). For the colors in PowerShell in general, that is a setting of the console. For Windows Terminal, you can choose a different [color scheme](https://docs.microsoft.com/windows/terminal/customize-settings/color-schemes), but winget will still use blue (or rainbow).

**Reply by Tiger348** [reply]
Thank you for your time when providing guidence ...with your knowledge could you explain why my system displays the progressBar colour in the dark grey in previous screen shot when in Terminal console or white when I use PowerShell in WIN10?
Thnak you in advance,

Tiger348

**Reply by florelis** [reply]
I think the progress bar is set by winget to the system accent color (I was wrong before when I said it was blue). Is your system accent color set to white/gray?

I think we currently don't have a way to set the progress bar color to a specific color, but you can open an issue requesting the feature if it's something you would like to see :)

**Reply by Tiger348** [reply]
Hi lechacon,
I found this article online...does it have any merrit to solving the progressBar challenge
Happy to have your thoughts.
Tiger348
![ProgressBar PowerShell Idea](https://user-images.githubusercontent.com/109897924/182545152-c62bc876-89d7-4fd8-a217-99a607100d78.png)

**Reply by florelis** [reply]
That is about the native progress bar in PowerShell. It's the one that shows up at the top of the screen, usually green I believe.
winget.exe does not use that progress bar; but a custom one that also works outside of PowerShell.

**Comment by Table-09** [comment]
I am having a similar problem but this isn't working. I did the environmental variable changes as well but to no avail. 

![image](https://user-images.githubusercontent.com/79593307/182406923-1e10affe-81f7-4a9d-a00f-13c6da0b9b22.png)
![image](https://user-images.githubusercontent.com/79593307/182406962-151bcba1-c93d-4dde-844d-1d8ce1f85a07.png)
![image](https://user-images.githubusercontent.com/79593307/182407321-c12b98af-4d24-4cec-bf52-7af8e6ba061b.png)
![image](https://user-images.githubusercontent.com/79593307/182407382-c86d47d5-dbb6-4aba-96c1-392cff4cad68.png)


Works in powershell but not cmd, this is driving me crazy. I even copied winget files from a brand new install after the powershell version stopped working, it allowed me (after not being recognized in either ps or cmd) to use it in PS after doing an in-place install, copying winget files from new computer and force-adding the app-pkg via powershell

**Reply by florelis** [reply]
Did you already look at the [troubleshooting guide](https://github.com/microsoft/winget-cli/blob/master/doc/troubleshooting/README.md)?

If that doesn't help, feel free to open a new Issue about this. I think it would be more likely that someone able to help you sees it if it was in an issue instead of a discussion.

**Reply by Table-09** [reply]
I already opened a new issue and someone pointed me to go to this one. seems like nobody knows what the hell is going on with this issue though.

**Reply by florelis** [reply]
Sorry, I hadn't seen #2391. I still don't know what the issue is, but I left a comment with some suggestions to try to narrow it down.
